src/azure-account.ts

```diff
@@ -387,10 +387,23 @@ export class AzureLoginHelper {
 						ignoreFocusOut: true
 					});
 					if (armUrl) {
-						await config.update(customCloudArmUrlKey, armUrl, getCurrentTarget(config.inspect(customCloudArmUrlKey)));
+						const tenantId = await window.showInputBox({
+							prompt: localize('azure-account.enterTenantId', "Enter the Tenant Id"),
+							placeHolder: "Tenant Id..",
+							ignoreFocusOut: true
+						});
+						if (tenantId) {
+							await config.update(customCloudArmUrlKey, armUrl, getCurrentTarget(config.inspect(customCloudArmUrlKey)));
+							await config.update('tenant', tenantId, getCurrentTarget(config.inspect('tenant')));
+						} else {
+							return;
+						}
 					} else {
 						return;
 					}
+				} else {
+					// Set tenant Id back to commonTenantId if user isn't using Azure Custom Cloud to avoid unmatched tenant id issue.
+					await config.update('tenant', commonTenantId, getCurrentTarget(config.inspect('tenant')));
```

Comment:
I'm wary of reseting this - I don't think we want to clear this. A user could have set a tenant setting and be using a different cloud

Reply:
For this piece of code, you can see from its context https://github.com/microsoft/vscode-azure-account/blob/917fdc42cd761e63ea045a8d38d7a5cbac369cd9/src/azure-account.ts#L382-L407
It will only be called when the cloud user selects is different from the one we have in config and that cloud can not be Azure Custom Cloud. 
A user could set a tenant id can be used for serval cloud. But sometime the tenant id won't be same, for example, the tenant id of Azure Custom Cloud may not work for the other clouds. For example, if we try to sign in to Azure if we have Azure Custom Cloud tenant id in our setting. Error as below will be shown. 
![image](https://user-images.githubusercontent.com/73665970/112708273-40d24180-8e6e-11eb-94c8-ff41b004ba38.png)
That is why I reset tenant id here. 
Will clouds like Azure, AzureChina... have specific tenant id other than common? If so, I think we probably could make the tenant inputbox available every time user selects a cloud which is different from the one we are having in config, not only when user is selecting Azure Custom Cloud. What do you think, Rachel? 
@bganapa, could you also take a look and give us some feedback if you have some idea about this.

Reply:
Yeah, for other clouds users can potentially configure their own tenant id. I think it would make sense to show the tenant input box every time the user chooses a different cloud so they can change it to the correct value. I think the majority of the time for the other clouds they would be using `common`, though, so it would be nice if we prefilled the input box with `common` then and left it empty for Azure Custom Cloud.